### PR TITLE
Providers in extras are properly configured and verified

### LINKED ISSUE #12255: Statsd (tbc. if more extras) tries to install provider package where it is missing
**Apache Airflow version**: 2.0.0b2

**What happened**:

Starting in 2.0.0b2, extras that aren't also providers like `statsd` prevents Airflow from being installed.

**How to reproduce it**:
```
$ docker run --rm -ti python:3.6 bash
# pip install apache-airflow[statsd]==2.0.0b2
Collecting apache-airflow[statsd]==2.0.0b2                 
  Downloading apache_airflow-2.0.0b2-py3-none-any.whl (4.5 MB)                    
...
Collecting statsd<4.0,>=3.3.0; extra == "statsd"          
  Downloading statsd-3.3.0-py2.py3-none-any.whl (11 kB)
ERROR: Could not find a version that satisfies the requirement apache-airflow-providers-statsd; extra == "statsd" (from apache-airflow[statsd]==2.0.0b2) (from versions: none)         
ERROR: No matching distribution found for apache-airflow-providers-statsd; extra == "statsd" (from apache-airflow[statsd]==2.0.0b2)
```

I believe this is from https://github.com/apache/airflow/pull/12233

### PR DESCRIPTION
**Pull request by potiuk**
This fixes #12255 - where we published beta2 release with some
extras pulling non-existing providers.

The exact list of providers that had problems:

Wrongly named extras/providers:

* apache.presto: it was badly named -> renamed to 'presto'
* spark (badly pointing to spark instead of apache.spark)
* yandexcloud (the name remains there but we've also added 'yandex' extra to correspond 1-1 with 'yandex' provider

Extras that were wrongly marked as having providers, where they had
none:

* dask
* rabbitmq
* sentry
* statsd
* tableau
* virtualenv

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
cc: @jedcunningham @mjpieters -> there were few more :(

File: scripts/ci/pre_commit/pre_commit_check_extras_have_providers.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+
+def check_all_providers() -> List[str]:
+    errors: List[str] = []
+    for extra, providers in EXTRAS_PROVIDERS_PACKAGES.items():
+        for provider in providers:
+            provider_directory = get_provider_directory(provider)
+            if not os.path.isdir(provider_directory):
+                errors.append(
+                    f"The {extra} has provider {provider} that has" f" missing {provider_directory} directory"
```
Comment (kaxil) [inline scripts/ci/pre_commit/pre_commit_check_extras_have_providers.py]:
```suggestion
                    f"The {extra} has provider {provider} that has missing {provider_directory} directory"
```

File: scripts/ci/pre_commit/pre_commit_check_extras_have_providers.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                    f" missing README.md in the {provider_directory} directory"
+                )
+    return errors
+
+
+if __name__ == '__main__':
+    errors = check_all_providers()
+    if errors:
+        for message in errors:
+            print(f"{message}", file=sys.stderr)
```
Comment (kaxil) [inline scripts/ci/pre_commit/pre_commit_check_extras_have_providers.py]:
```suggestion
            print(message, file=sys.stderr)
```

File: scripts/ci/pre_commit/pre_commit_check_extras_have_providers.py (hunk 1 above)
Comment (potiuk) [reply scripts/ci/pre_commit/pre_commit_check_extras_have_providers.py]: Ah. Blacks's joining the lines :)

File: scripts/ci/pre_commit/pre_commit_check_extras_have_providers.py (hunk 2 above)
Comment (potiuk) [reply scripts/ci/pre_commit/pre_commit_check_extras_have_providers.py]: Oh yeah. It was more complex before. I have not noticed how much i simplified the f-string :)

**Comment by potiuk** [comment]
Fixes applied!

**Review by kaxil — APPROVED** [review]

**Pull request merged**